web/src/lib/static/admin/findDeadLinks.ts

```diff
@@ -238,29 +265,42 @@ export const findDeadLinks = async (): Promise<Record<string, string[]>> => {
     );
   };
 
-  for (const page of pages) {
-    const promise = new Promise((resolve) => {
-      const htmlUrlChecker = new HtmlUrlChecker(
-        {},
-        {
-          link: (result: any): void => {
-            if (result.broken && !isTemplateLink(result.url.original)) {
-              deadLinks[page].push(result.url.original);
-            }
-          },
-          end: (): void => {
-            resolve({});
-          },
-        }
-      );
-      const url = new URL(process.env.REDIRECT_URL || "");
-      htmlUrlChecker.enqueue(`${url.origin}${page}`, {});
-    });
+  const numberOfBatches = Math.ceil(pages.length / 5);
+
+  for (let j = 0; j < numberOfBatches; j++) {
+    const pagePromises = [];
+    const startIndex = j * 5;
+    const batch = pages.slice(startIndex, startIndex + 5);
```

Comment:
I have a vague memory of you mentioning batching in standup. What was the reasoning for this again? Just curious.

Reply:
So the reasoning is that we believe that starting jobs is faster than finishing jobs so what was happening is it would eventually get to the point where it was trying to do like 200 job simultaneously which ends up tanking the computer